web/src/settings_realm_user_settings_defaults.js

```diff
@@ -62,6 +67,25 @@ export function set_up() {
         });
     });
 
+    if (!page_params.development_environment) {
+        $("#realm_dense_mode").on("change", (e) => {
+            const val = $(e.target).prop("checked");
+            if (val) {
+                $container.find(".information-density-settings").hide();
+                return;
+            }
+
+            if (
+                !realm_user_settings_defaults.dense_mode &&
+                (realm_user_settings_defaults.web_font_size_px !== NON_COMPACT_MODE_FONT_SIZE_PX ||
+                    realm_user_settings_defaults.web_line_height_percent !==
+                        NON_COMPACT_MODE_LINE_HEIGHT_PERCENT)
+            ) {
+                $container.find(".information-density-settings").show();
```

Comment:
There are still some cases where the error is returned from the API like -
- initially the setting is set to something other than 14/122 and 16/140 combination. So, the inputs are shown.
- The setting is changed to something other than 14/122.
- And then the "Compact mode" checkbox is checked. The inputs will hide, but the request data will still contain the new values from inputs and clicking save will result in an error from API.

One way here can be to set the inputs to 14/122 while hiding the inputs. But then if I uncheck the "Compact mode" checkbox, the inputs will be shown but they cannot be set to the values that I changed them to before checking "Compact mode" checkbox since we do not store that value. We can store that intermediate value as well, but that would make the code confusing and hard to read. It can instead be changed to their actual value in DB. If we are fine with this then we can do that.

Reply:
I don't think it's worth it given the low frequency with which we expect this to see interaction. But thanks for flagging.